src/utils/archive.ts

```diff
@@ -200,6 +201,72 @@ export async function createPublicDrive(
   };
 }
 
+/**
+ * Sends the ardrive fee based on the AR price of data upload
+ *
+ * @param arweave Arweave client
+ * @param keyfile Wallet keyfile
+ * @param arPrice Cost of data upload
+ *
+ * @returns Fee Transaction if successful
+ */
+export async function sendArDriveFee(
+  keyfile: JWKInterface,
+  arPrice: number,
+  arweave: Arweave
+) {
+  try {
+    // Return if upload cost is zero, this should likely never happen
+    if (arPrice <= 0) {
+      return;
+    }
+
+    // If the fee is too small, we assign a minimum
+    let fee = Math.max(
+      arPrice * ((await getArDriveTipPercentage()) / 100),
```

Comment:
No need to divide by 100 now that you're always returning decimal representations of percentages.

Reply:
[Resolved](https://github.com/th8ta/ArConnect/pull/33/commits/66b11719315a0ac4ea310ba6d165889efd900a59)